src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs

```diff
@@ -44,14 +46,49 @@ public DefaultValueAttribute(Type type, string value)
                 {
                     _value = TimeSpan.Parse(value);
                 }
-                else
+                else if (type.Module == typeof(string).Module)
                 {
                     _value = Convert.ChangeType(value, type, CultureInfo.InvariantCulture);
                 }
+                else
+                {
+                    _value = ConvertFromInvariantString(type, value);
+                }
             }
             catch
             {
             }
+
+            object ConvertFromInvariantString(Type typeToConvert, string stringValue)
```

Comment:
nit: missing private

Reply:
It's a local func...if i'm not mistaken it's not possibile add access modifier

Reply:
I see, totally missed the placement of the closing brace.   Not sure I've seen any usage of local functions yet in our codebase and so far our [style guidelines are mum on it](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md).  @jkotas do you have any opinion on style guidelines for local functions?  Normally I'd ask @stephentoub @danmosemsft @weshaggard but they're all out.

Just some initial thoughts on style: 
place local functions at the end of the parent member and an explicit return before the local function to make it visually clear where the parent member ends.
include a comment above the local function.

Reply:
I used a local func because we are replacing simple TypeConverter call(with a lot of code) and it's used only in this method(ctor), to me it's more clear...but let me know if i need to tranform to normal private method!

Reply:
> Not sure I've seen any usage of local functions yet in our codebase

I know people have been adding them here and there for a while. Here is an example: 
https://github.com/dotnet/corefx/pull/24389/files#diff-8df21afb33a350b9ca945d04a5296b9bR119. The style is about what described.

Reply:
done